geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -3138,7 +3138,7 @@ public Pool determineDefaultPool(PoolFactory poolFactory) {
     invokeRegionAfter(region);
 
     // Added for M&M . Putting the callback here to avoid creating RegionMBean in case of Exception
-    if (!region.isInternalRegion()) {
+    if (region.isRegionRegistrationRequireOnJmx()) {
```

Comment:
As a community we are trying to refactor code and separate concerns. This change is going in the wrong direction -- you're combining JMX concerns into Cache and Region which is beyond the scope of these classes.